### DISCUSSION TITLE
need help parsing i3-msg

### DISCUSSION THREAD

**Discussion by internet-police** [i3/i3#5537] [Q&A]
Heya!
can anyone pls help silly me to use "i3-msg -t get_tree  |  jq"
so it outputs a list of every window with its class and id

i basically want to pipe this into fzf, and then focus the selected window
but i have some trouble understanding the jq command

**Comment by okraits** [comment]
You could also try https://github.com/davatorium/rofi. It's a window switcher and launcher similar to what you are trying to achieve.

**Comment by orestisfl** [comment]
The docs for the IPC interface are here: https://i3wm.org/docs/ipc.html. E.g. see [the `get_tree` reply](https://i3wm.org/docs/ipc.html#_tree_reply)

Other than figuring out the raw response, you could use an existing ipc library, there is a list here: https://i3wm.org/docs/ipc.html#_see_also_existing_libraries
